ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +71,50 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
-    top: Math.floor(heightWithMargins * pageIndex + marginTopPixels + (height - bottomPixels)),
+  const pageId = generatePageIdFromIndex(pageIndex);
+
+  const pageIdElement = document.getElementById(pageId);
+
+  if (!pageIdElement) {
+    return;
+  }
+
+  const parentElement = getScrollParent(pageIdElement);
+
+  if (!parentElement) {
+    return;
+  }
+
+  parentElement.scrollTo({
+    top: calculateTop(heightWithMargins, pageIndex, marginTopPixels, height, bottomPixels),
     left: Math.floor(leftPixels),
     behavior: 'smooth',
   });
 }
 
+export function getScrollParent(node: HTMLElement): HTMLElement | null {
+  if (typeof document === 'undefined') {
+    return null;
+  }
+  if (!node) {
+    return document.documentElement;
+  }
+  if (node.scrollHeight > node.clientHeight) {
+    return node;
+  }
+  return getScrollParent(node.parentNode as HTMLElement);
+}
+
+export function calculateTop(
```

Comment:
It's worth adding a unit to the name. Something like `calculateTopPx`. I don't think you need `Pixels` in the variable names either, since `px` is pretty well known in the JS world.